File: Search/LinearSearch/Ruby/linear_search_test.rb
require 'minitest/autorun'
require_relative 'linear_search'


describe "linear_search" do
  TEST_DATA = [1, 2, 3, 4, 5, 6]

  it "returns the position of the given value in the given list" do
    expect(liner_search(TEST_DATA, 3)).must_equal(2)
  end

  it "raises NoSuchValueError when the given value dose not exist in the list" do
    expect{ liner_search(TEST_DATA, 100) }.must_raise(NoSuchValueError)
  end
end

File: Search/LinearSearch/TypeScript/linearSearch.test.ts
import { linearSearch, newLnearSearch } from './linearSearch'

describe('linerSearch', () => {
  it('returns true if the given value exists in the given list', () => {
    const data = [2, 3, 4, 5, 6, 7, 8, 9]
    expect(linearSearch(data, 6)).toBeTruthy()
  })

  it('returns false if the given value dose not exist in the given list', () => {
    const data = [2, 3, 4, 5, 6, 7, 8, 9]
    expect(linearSearch(data, 10)).toBeFalsy()
  })

  it('returns the position of the given value in the given list', () => {
    const data = [2, 3, 4, 5, 6, 7, 8, 9]
    expect(newLnearSearch(data, 6)).toEqual(4)
  })

  it('raises an error when the given value dose not exist in the given list', () => {
    const data = [2, 2, 2]
    expect(() => {
      newLnearSearch(data, 1)
    }).toThrow('No such value')
  })
})
